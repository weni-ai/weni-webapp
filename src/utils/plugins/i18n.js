import Vue from 'vue';
import VueI18n from 'vue-i18n';

import en from '../../locales/en.json';
import es from '../../locales/es.json';
import pt_br from '../../locales/pt_br.json';

Vue.use(VueI18n);

const languages = {
  'pt-br': pt_br,
  'pt-BR': pt_br,
  en: en,
  es: es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en',
  fallbackLocale: 'en',
  messages,
});

export default i18n;
