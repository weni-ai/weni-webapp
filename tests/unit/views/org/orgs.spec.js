import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Router from 'vue-router';
import Projects from '@/views/org/orgs.vue';
import i18n from '@/utils/plugins/i18n';
import OrgList from '@/components/orgs/orgList.vue';
import { org } from '../../../__mocks__';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Router);

const router = new Router();

jest.mock('@/services/SecurityService.js', () => {});
jest.mock('@/api/request.js', () => {});

describe('orgs.vue', () => {
  let wrapper;
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      clearCurrentOrg: jest.fn(),
      clearCurrentProject: jest.fn(),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(Projects, {
      localVue,
      i18n,
      store,
      router,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        RouterLink: RouterLinkStub,
        OrgList,
        SkeletonLoading: true,
        UnnnicButton: true,
        UnnnicIconSvg: true,
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // SHOULD WE TEST IT AFTER.
  // describe('tryAgain()', () => {
  //   it('should call the function tryAgain', () => {
  //     const spy = jest.spyOn(wrapper.vm.$refs.orgList, 'reloadOrganizations');
  //     expect(spy).not.toHaveBeenCalled();
  //     wrapper.vm.tryAgain();
  //     expect(spy).toHaveBeenCalledTimes(1);
  //   });
  // });

  describe('organizationsStatus() watcher', () => {
    it('Should set error true', () => {
      wrapper.vm.organizationsStatus = 'error';
      wrapper.vm.$nextTick(() => {
        expect(wrapper.vm.error).toBe(true);
      });
    });

    it('Should test loaded status for error', () => {
      wrapper.vm.organizationsStatus = 'loaded';
      wrapper.vm.$nextTick(() => {
        expect(wrapper.vm.error).toBe(false);
      });
    });

    it('Should test empty status for error', () => {
      wrapper.vm.organizationsStatus = 'empty';
      wrapper.vm.$nextTick(() => {
        expect(wrapper.vm.error).toBe(false);
      });
    });
  });
});
