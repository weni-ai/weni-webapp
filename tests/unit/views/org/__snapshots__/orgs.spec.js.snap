// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orgs.vue should be rendered properly 1`] = `
<div class="weni-orgs">
  <div class="unnnic-grid-lg">
    <div class="weni-orgs__left unnnic-grid-span-5">
      <div class="box aux-blue weni-orgs__left__icon">
        <unnniciconsvg-stub icon="building-2-1" size="xl" scheme="aux-blue"></unnniciconsvg-stub>
      </div>
      <h1>some specific text</h1>
      <p>
        some specific text
      </p> <a>
        <unnnicbutton-stub size="large" type="secondary" iconleft="add-1">
          some specific text
        </unnnicbutton-stub>
      </a>
    </div>
    <div class="unnnic-grid-span-2"></div>
    <div class="unnnic-grid-span-5 weni-orgs__right">
      <div class="weni-orgs__list">
        <div class="weni-org-list__wrapper list-container">
          <div class="weni-org-list">
            <new-infinite-loading-stub>
              <div style="display: flex;">
                <div style="flex: 1; margin-right: 1rem;">
                  <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="100%" height="32px"></unnnic-skeleton-loading-stub>
                  <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="100%" height="19px"></unnnic-skeleton-loading-stub>
                  <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="100%" height="19px" style="margin-bottom: 1rem;"></unnnic-skeleton-loading-stub>
                  <div style="display: flex;">
                    <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="64px" height="24px" style="margin-right: 0.5rem;"></unnnic-skeleton-loading-stub>
                    <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="211px" height="24px"></unnnic-skeleton-loading-stub>
                  </div>
                </div>
                <div style="display: flex; align-items: center;">
                  <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="73px" height="38px" style="margin-right: 0.5rem;"></unnnic-skeleton-loading-stub>
                  <unnnic-skeleton-loading-stub prefix="unnnic" count="1" duration="1.5" tag="div" width="17px" height="38px"></unnnic-skeleton-loading-stub>
                </div>
              </div>
            </new-infinite-loading-stub>
          </div>
          <right-side-bar-stub type="change-name" props="[object Object]"></right-side-bar-stub>
          <right-side-bar-stub type="view-members" props="[object Object]"></right-side-bar-stub>
          <right-side-bar-stub type="manage-members" props="[object Object]"></right-side-bar-stub>
        </div>
      </div>
    </div>
  </div>
  <skeletonloading-stub style="display: none;"></skeletonloading-stub>
</div>
`;
